### PR TITLE
Ps 5.7 myr 145

### PR DESCRIPTION
**Pull request by george-lorch**

### CODE REVIEW

**Comment by george-lorch** [comment]
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1081/

Some unstable test regressions introduced and logged in MYR and upstream for tracking.

File: mysql-test/suite/rocksdb.sys_vars/t/rocksdb_compact_cf_basic.test [hunk 1]
```diff
@@ -1,3 +1,6 @@
+
+call mtr.add_suppression(" Column family '[a-z]*' not found.");
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb.sys_vars/t/rocksdb_compact_cf_basic.test]: Should go after the have_foo checks (not necessarily actionable in the current PR)

File: mysql-test/suite/rocksdb/include/bloomfilter.inc [hunk 2]
```diff
@@ -26,12 +26,20 @@ DELIMITER ;//
 --source $ddl
 --source suite/rocksdb/include/bloomfilter_load_select.inc
 
+--exec sed s/##CF##/" COMMENT 'rev:cf_short_prefix'"/g $tmpl_ddl > $ddl
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/include/bloomfilter.inc:29]: SQL string statement manipulation or perl would be more canonical for portability, but I guess it's not something to lose sleep over

File: mysql-test/suite/rocksdb/t/partition.test [hunk 3]
```diff
@@ -28,7 +34,677 @@ SHOW TABLES;
...
+  PARTITION BY RANGE(YEAR(joined)) (
+    PARTITION p0 VALUES LESS THAN (1960),
+    PARTITION p1 VALUES LESS THAN (1970),
+    PARTITION p2 VALUES LESS THAN (1980),
+    PARTITION p3 VALUES LESS THAN (1990),
+    PARTITION p4 VALUES LESS THAN MAXVALUE
+  );
+
+# Partition names are not case-sensitive.
+--error 1517
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/partition.test]: ER_SAME_NAME_PARTITION

File: mysql-test/suite/rocksdb/t/partition.test [hunk 4]
```diff
@@ -28,7 +34,677 @@ SHOW TABLES;
...
+    PARTITION p1 VALUES LESS THAN (10,20,100),
+    PARTITION p2 VALUES LESS THAN (10,30,50),
+    PARTITION p3 VALUES LESS THAN (MAXVALUE,MAXVALUE,MAXVALUE)
+    );
+
+SELECT (0,25,50) < (10,20,100), (10,20,100) < (10,30,50);
+
+-- ERROR 1493 (HY000): VALUES LESS THAN value must be strictly increasing for each partition
+
+--error 1493
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/partition.test]: ER_RANGE_NOT_INCREASING_ERROR and then the comment above becomes redundant

File: mysql-test/suite/rocksdb/t/partition.test [hunk 5]
```diff
@@ -28,7 +34,677 @@ SHOW TABLES;
...
+        SUBPARTITION s2,
+        SUBPARTITION s3
+      ),
+      PARTITION p2 VALUES LESS THAN MAXVALUE (
+        SUBPARTITION s4,
+        SUBPARTITION s5
+      )
+    );
+
+--error 1064
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/partition.test]: ER_PARSE_ERROR

File: mysql-test/suite/rocksdb/t/partition.test [hunk 6]
```diff
@@ -28,7 +34,677 @@ SHOW TABLES;
...
+  PARTITION BY RANGE( YEAR(dob) ) (
+    PARTITION p0 VALUES LESS THAN (1970),
+    PARTITION p1 VALUES LESS THAN (1980),
+    PARTITION p2 VALUES LESS THAN (1990)
+  );
+
+ALTER TABLE members_3 ADD PARTITION (PARTITION p3 VALUES LESS THAN (2000));
+
+# ERROR 1493 (HY000): VALUES LESS THAN value must be strictly increasing for each partition
+--error 1493
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/partition.test]: Same as above

File: mysql-test/suite/rocksdb/t/partition.test [hunk 7]
```diff
@@ -28,7 +34,677 @@ SHOW TABLES;
...
+  id INT,
+  fname VARCHAR(30),
+  lname VARCHAR(30),
+  signed DATE
+) ENGINE=ROCKSDB
+  PARTITION BY LINEAR KEY(signed)
+  PARTITIONS 12;
+
+# ERROR 1508 (HY000): Cannot remove all partitions, use DROP TABLE instead
+--error 1508
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/partition.test]: ER_DROP_LAST_PARTITION and then the comment above disappears

File: storage/rocksdb/README [hunk 8]
```diff
@@ -34,5 +42,9 @@ Several refinements:
      "_vect" for a std::vector etc.
 
 == Running Tests ==
-To run tests from rocksdb, rocksd_rpl or rocksdb_sys_vars packages, use the following parameters:
---mysqld=--default-storage-engine=rocksdb  --mysqld=--skip-innodb --mysqld=--default-tmp-storage-engine=MyISAM --mysqld=--rocksdb
\ No newline at end of file
+To run tests from rocksdb, rocksdb_rpl or other rocksdb_* packages, use the
+following parameters:
+  --default-storage-engine=rocksdb
```
Comment (laurynas-biveinis) [inline storage/rocksdb/README]: This is not correct for us, is it?

File: storage/rocksdb/ha_rocksdb.h [hunk 9]
```diff
@@ -126,6 +127,34 @@ const char *const BG_THREAD_NAME = "myrocks-bg";
 const char *const INDEX_THREAD_NAME = "myrocks-index";
 
 /*
+  Separator between partition name and the qualifier. Sample usage:
+
+  - p0_cfname=foo
+  - p3_tts_col=bar
+*/
+const char RDB_PER_PARTITION_QUALIFIER_NAME_SEP = '_';
```
Comment (laurynas-biveinis) [inline storage/rocksdb/ha_rocksdb.h]: constexpr? Not necessarily actionable in this PR

File: storage/rocksdb/ha_rocksdb.h [hunk 10]
```diff
@@ -160,6 +189,11 @@ const char *const INDEX_THREAD_NAME = "myrocks-index";
 #define MAX_SUBCOMPACTIONS 64
 
 /*
+  Default value for rocksdb_sst_mgr_rate_bytes_per_sec = 64 MB.
+*/
+#define DEFAULT_SST_MGR_RATE_BYTES_PER_SEC (64 * 1024 * 1024)
```
Comment (laurynas-biveinis) [inline storage/rocksdb/ha_rocksdb.h]: Likewise

File: storage/rocksdb/rdb_sst_info.h [hunk 11]
```diff
@@ -21,46 +21,97 @@
...
+                 const rocksdb::DBOptions &db_options, const std::string &name,
+                 const bool tracing);
+    ~Rdb_sst_file();
+
+    rocksdb::Status open();
+    rocksdb::Status put(const rocksdb::Slice &key, const rocksdb::Slice &value);
+    rocksdb::Status commit();
+
+    inline const std::string get_name() const { return m_name; }
+    inline int compare(rocksdb::Slice key1, rocksdb::Slice key2) {
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_sst_info.h]: const method?

File: storage/rocksdb/rdb_sst_info.h [hunk 12]
```diff
@@ -21,46 +21,97 @@
...
+    explicit Rdb_sst_stack(size_t max_size)
+        : m_buffer(nullptr), m_buffer_size(max_size) {}
+    ~Rdb_sst_stack() { delete[] m_buffer; }
+
+    void reset() { m_offset = 0; }
+    bool empty() { return m_stack.empty(); }
+    void push(const rocksdb::Slice &key, const rocksdb::Slice &value);
+    std::pair<rocksdb::Slice, rocksdb::Slice> top();
+    void pop() { m_stack.pop(); }
+    size_t size() { return m_stack.size(); }
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_sst_info.h]: const method

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Minor changes only, not all of them actionable in this PR

**Comment by george-lorch** [comment]
Addressed most of the comments, jenkins re-spin here with another instability that I am thinking may be specific to jenkins slowness http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1085/

**Review by laurynas-biveinis — APPROVED** [review]
LGTM - @georgelorchpercona , do you want me to merge this while you are away?

**Pull request merged**